core/eolearn/tests/test_core_tasks.py

```diff
@@ -199,46 +199,28 @@ def test_remove_fails(patch: EOPatch) -> None:
         RemoveFeatureTask((FeatureType.BBOX, None))(patch)
 
 
-def test_duplicate_feature(patch):
-    mask_data = np.arange(10).reshape(5, 2, 1, 1)
-    feature_name = "MASK1"
-    duplicate_name = "MASK2"
-
-    patch = AddFeatureTask((FeatureType.MASK, feature_name))(patch, mask_data)
+@pytest.mark.parametrize(
+    "feature_specification",
+    [
+        [(FeatureType.DATA, "bands", "bands2")],
+        [(FeatureType.DATA, "bands", "bands2"), (FeatureType.MASK_TIMELESS, "mask", "mask2")],
+        [(FeatureType.DATA, "bands", f"bands{i}") for i in range(5)],
+    ],
+)
+@pytest.mark.parametrize("deep", [True, False])
+def test_duplicate_feature(feature_specification: List[FeaturesSpecification], deep: bool, patch: EOPatch) -> None:
+    patch = DuplicateFeatureTask(feature_specification, deep)(patch)
 
-    duplicate_task = DuplicateFeatureTask((FeatureType.MASK, feature_name, duplicate_name))
-    patch = duplicate_task(patch)
+    for f_type, f_name, f_dup_name in feature_specification:
+        assert f_dup_name in patch[f_type]
+        assert (id(patch[(f_type, f_name)]) == id(patch[(f_type, f_dup_name)])) != deep
```

Comment:
no

if you want to have both in the same test, then do something like
```python
original_id = id(patch[(ftype, fname)])
duplicated_id = id(patch[(ftype, fname)])
assert original_id != duplicated_id if deep else original_id == duplicated_id
```

Reply:
done

Reply:
done